src/en/manhwabuddy/src/eu/kanade/tachiyomi/extension/en/manhwabuddy/ManhwaBuddy.kt

```diff
@@ -0,0 +1,138 @@
+package eu.kanade.tachiyomi.extension.en.manhwabuddy
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class ManhwaBuddy : ParsedHttpSource() {
+    override val baseUrl = "https://manhwabuddy.com/"
+    override val lang = "en"
+    override val name = "ManhwaBuddy"
+    override val supportsLatest = true
+
+    override fun chapterFromElement(element: Element): SChapter {
+        return SChapter.create().apply {
+            setUrlWithoutDomain(element.attr("abs:href"))
+            name = element.selectFirst(".chapter-name")!!.text()
+            date_upload = element.selectFirst(".ct-update")?.text()?.let { parseDate(it) }!!
+        }
+    }
+
+    private fun parseDate(dateStr: String): Long {
+        return try {
+            dateFormat.parse(dateStr)!!.time
+        } catch (_: ParseException) {
+            0L
+        }
+    }
+
+    private val dateFormat by lazy {
+        SimpleDateFormat("dd MMMM yyyy", Locale.ENGLISH)
+    }
+
+    override fun chapterListSelector(): String {
+        return ".chapter-list a"
+    }
+
+    override fun imageUrlParse(document: Document): String {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            setUrlWithoutDomain(element.selectFirst("a")!!.attr("abs:href"))
+            title = element.selectFirst("h4")!!.text()
+            thumbnail_url = element.selectFirst("img")!!.attr("src")
+        }
+    }
+
+    override fun latestUpdatesNextPageSelector(): String? {
+        return ".next"
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/page/$page", headers)
+    }
+
+    override fun latestUpdatesSelector(): String {
+        return ".latest-list .latest-item"
+    }
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            val info = document.selectFirst(".main-info-right")!!
+            author = info.selectFirst("li:contains(Author(s)) a")?.text()
+            status = when (info.selectFirst("li:contains(Status) span")?.text()) {
+                "Ongoing" -> SManga.ONGOING
+                "Complete" -> SManga.COMPLETED
+                else -> SManga.UNKNOWN
+            }
+            artist = info.selectFirst("li:contains(Artist(s)) a")?.text()
```

Comment:
Does Jsoup properly tokenize the selector here? To be on the safe side, consider:
```css
li:contains(Artist) a
```